Sources/SwiftGenKit/Parsers/Parser.swift

```diff
@@ -25,12 +40,31 @@ public protocol Parser {
 }
 
 public extension Parser {
+  /// Factory method for safely creating a parser instance. Options will be checked against the list of
+  /// supported options (in `allOptions`).
+  ///
+  /// - Parameter options: Dictionary of options, checked against `allOptions`.
+  /// - Parameter warningHandler: Callback for logging issues.
+  /// - Returns: An instance of the parser.
+  static func createWith(options: [String: Any], warningHandler: MessageHandler?) throws -> Self {
+    try allOptions.check(options: options)
```

Comment:
Not sure I like the factory method idea. I mean, what's the harm in doing that call in `init` and let consumers call `init` (instead of forgetting there's a factory method `createWith` they'd have to call instead of `init`)?

I don't think checking the options in `init` will have any significant performance effect or whatever, so not sure why you were reluctant to do it there, did I miss something?

Reply:
The thing is we'd need to repeat this (and not forget it) in each and every parser init.

Reply:
Suggestion: https://github.com/SwiftGen/SwiftGen/pull/597#discussion_r255347946